troposphere/static/js/components/images/common/SecondaryImageNavigation.react.js

```diff
@@ -27,10 +27,13 @@ define(function (require) {
           allImages = stores.ImageStore.getAll(),
           images = stores.ImageStore.fetchWhere({
             created_by__username: profile.get('username')
-          }) || [], 
-          favoritedImages = stores.ImageBookmarkStore.getBookmarkedImages() || [];
+          }) || [];
 
-      if(!images || !favoritedImages){
+      // only attempt to get bookmarks if there is a profile that might have them ...
+      var userLoggedIn = !!(profile && profile.get('selected_identity')),
+        favoritedImages =  userLoggedIn ? stores.ImageBookmarkStore.getBookmarkedImages() : [];
```

Comment:
So, the `Master.react` component is setting `context.profile` to the result of `ProfileStore.get()` in `getInitialState()`:

```javascript
    getInitialState: function () {
      var profile = stores.ProfileStore.get();
      if(!context.profile && profile) {
          context.profile = profile;
      }
      return this.getState();
    }
```
source: [troposphere/static/js/components/Master.react.js](https://github.com/iPlantCollaborativeOpenSource/troposphere/blob/larping-loon/troposphere/static/js/components/Master.react.js#L28-L34)

I'm _not_ sure how I feel about this. This is a _*total*_ side-effect in `getInitialState()`. [:warning:](http://4.bp.blogspot.com/-Y3F7Z1L4oBE/VXY5GdH3-3I/AAAAAAAAENA/LEFZgvS0a34/s1600/It-is-what-it-is.gif)